editor/src/messages/tool/tool_messages/path_tool.rs

```diff
@@ -1273,6 +1219,37 @@ impl Fsm for PathToolFsmState {
 
 				PathToolFsmState::Dragging(tool_data.dragging_state)
 			}
+			(PathToolFsmState::Ready, PathToolMessage::PointerMove { lock_angle, .. }) => {
+				let lock_angle_state = input.keyboard.get(lock_angle as usize);
+				if lock_angle_state {
+					tool_data.delete_segment_pressed = true;
+				} else {
+					tool_data.delete_segment_pressed = false;
+				}
```

Comment:
It would be much simpler to simply assign: `tool_data.delete_segment_pressed = lock_angle_state`.